app/models/registration_system.rb

```diff
@@ -81,4 +117,15 @@ def self.database_options
     }
   end
   private_class_method :database_options
+
+  def self.write_rhsm_config(params)
+    FileUtils.copy(RHSM_CONFIG_FILE, "#{RHSM_CONFIG_FILE}.miq_orig") unless File.exist?("#{RHSM_CONFIG_FILE}.miq_orig")
+    rhsm_config = File.read(RHSM_CONFIG_FILE)
+    rhsm_config[/(\s*)proxy_hostname(\s*)=(.*)/, 3] = " #{params[:proxy_hostname]}"
+    rhsm_config[/(\s*)proxy_port(\s*)=(.*)/, 3]     = " #{params[:proxy_port]}"
+    rhsm_config[/(\s*)proxy_user(\s*)=(.*)/, 3]     = " #{params[:proxy_user]}"
+    rhsm_config[/(\s*)proxy_password(\s*)=(.*)/, 3] = " #{params[:proxy_password]}"
```

Comment:
@jrafanie no. The code does not create the lines but it does not need to because the file is delivered with them but with no values. So the file is delivered with:
```
proxy_hostname =
proxy_prot =
proxy_user =
proxy_password =
```

The code `does` update any existing values.